compiler/rustc_error_messages/locales/en-US/parse.ftl

```diff
@@ -390,3 +390,72 @@ parse_where_clause_before_tuple_struct_body = where clauses are not allowed befo
     .name_label = while parsing this tuple struct
     .body_label = the struct body
     .suggestion = move the body before the where clause
+
+parse_unexpected_token_after_dot = unexpected token: `{$actual}`
+
+parse_expect_trait_found_type = expected a trait, found type
+
+parse_trait_alias_bound = bounds are not allowed on trait aliases
+
+parse_trait_alias_cannot = trait aliases cannot be `{$attribute}`
+    .label = {parse_trait_alias_cannot}
+
+parse_self_raw_pointer = cannot pass `self` by raw pointer
+    .label = {parse_self_raw_pointer}
+
+parse_cannot_be_raw_ident = `{$ident}` cannot be a raw identifier
+
+parse_cr_doc_comment = bare CR not allowed in {$block ->
+    [true] block doc-comment
+    *[false] doc-comment
+}
+
+parse_no_digits_literal = no valid digits found for number
+
+parse_invalid_digit_literal = invalid digit for a base {$base} literal
+
+parse_empty_exponent_float = expected at least one digit in exponent
+
+parse_float_literal_unsupported_base = {$base} float literal is not supported
+
+parse_unescape_error = {$err ->
+    [ZeroChars] empty character literal
+    [LoneSlash] invalid trailing slash in literal
+    [MultipleSkippedLinesWarning] multiple lines skipped by escaped newline
+    [EmptyUnicodeEscape] empty unicode escape
+    [OverlongUnicodeEscape] overlong unicode escape
+    [OutOfRangeHexEscape] out of range hex escape
+    [TooShortHexEscape] numeric character escape is too short
+    [BareCarriageReturnInRawString] bare CR not allowed in raw string
+    [UnicodeEscapeInByte] unicode escape in byte string
+    *[default] {$unreachable_ICE_if_reached}
```

Comment:
Since this error should only be emitted for these specific cases, here I've added an invalid variable so we ICE which allows us to spot the error, unlike some dummy text. I'm not sure if there's a better way to do this

Reply:
I don't think this is a good idea. You could use an enum like how is done here: https://github.com/rust-lang/rust/blob/master/compiler/rustc_ty_utils/src/errors.rs